app/pages/home-for-user/circle-ribbon.jsx

```diff
@@ -0,0 +1,233 @@
+import React from 'react';
+import SVGLink from './svg-link';
+
+import style from './circle-ribbon.styl';
+void style;
+
+let instanceCount = 0;
+
+function defaultHREFTemplate(project) {
+  return `#project=${project.id}`;
+}
+
+const CircleRibbon = React.createClass({
+  propTypes: {
+    size: React.PropTypes.string,
+    weight: React.PropTypes.number,
+    gap: React.PropTypes.number,
+    image: React.PropTypes.string,
+    loading: React.PropTypes.bool,
+    data: React.PropTypes.array,
+    hrefTemplate: React.PropTypes.func,
+    onClick: React.PropTypes.func,
+    user: React.PropTypes.object,
+  },
+
+  getDefaultProps() {
+    return {
+      size: '10em',
+      weight: 10,
+      gap: 2,
+      image: '//lorempixel.com/100/100/animals/1',
+      loading: false,
+      data: [],
+      hrefTemplate: defaultHREFTemplate,
+      onClick: () => {},
+    };
+  },
+
+  getInitialState() {
+    return {
+      totalClassifications: 0,
+      hoverIndex: -1,
+    };
+  },
+
+  componentDidMount() {
+    this.id = instanceCount;
+    this.point = this.refs.svg.createSVGPoint();
+    instanceCount += 1;
+    this.setTotal(this.props.data);
+  },
+
+  componentWillReceiveProps(nextProps) {
+    this.setTotal(nextProps.data);
```

Comment:
Ok, so I think what you'll want is to call `this.setTotal()` when `this.props.data` does not equal `nextProps.data`, so
```
if (this.props.data !== nextProps.data) { this.setTotal(nextProps.data) }
```
If you're not explicitly checking that, then it'll unnecessarily call `this.setTotal` whenever any new prop is received even if the data prop is still just an empty array.